### PR TITLE
Add board ST NUCLEO L412RB-P

### LINKED ISSUE #510: Add board ST Nucleo L412RB-P
PlatformIO supports ST Nucleo L412KB but the RB variant is slightly different. I've been working on adding this to Zephyr and want to try and add it to PlatformIO as well.

It is my understanding that this https://docs.platformio.org/en/latest/platforms/creating_board.html documentation explains how to add the board.

* Board: https://www.digikey.com/en/products/detail/stmicroelectronics/NUCLEO-L412RB-P/9656230
* User manual: [UM2206.pdf](https://github.com/platformio/platform-ststm32/files/6215621/dm00387966-stm32-nucleo64p-boards-mb1319-stmicroelectronics.1.pdf)

### PR DESCRIPTION
**Pull request by GunZi200**
Closes #510

### CODE REVIEW

File: boards/nucleo_l412rb_p.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "build": {
+    "core": "stm32",
```
Comment (valeros) [inline boards/nucleo_l412rb_p.json:3]: Redundant field, there is no Arduino support for this board.

File: boards/nucleo_l412rb_p.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    "cpu": "cortex-m4",
+    "extra_flags": "-DSTM32L412xx",
+    "f_cpu": "80000000L",
+    "mcu": "stm32l412rb",
```
Comment (valeros) [inline boards/nucleo_l412rb_p.json:7]: Please specify the full MCU item

File: boards/nucleo_l412rb_p.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    "product_line": "STM32L412xx",
+    "variant": "NUCLEO_L412RB_P"
```
Comment (valeros) [inline boards/nucleo_l412rb_p.json:9]: Redundant field, there is no Arduino framework for this board.

File: boards/nucleo_l412rb_p.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    "openocd_target": "stm32l4x",
+    "svd_path": "STM32L4x2.svd"
+  },
+  "frameworks": [
+    "arduino",
+    "cmsis",
+    "mbed",
+    "stm32cube",
+    "libopencm3",
+    "zephyr"
```
Comment (valeros) [inline boards/nucleo_l412rb_p.json:29]: It seems that this board can be used only with low-level frameworks `cmsis` and `stm32cube`.

**Review by valeros — CHANGES_REQUESTED** [review]

**Comment by valeros** [comment]
Hi @GunZi200 ! Thanks for the PR. Please see my comments above.

File: boards/nucleo_l412rb_p.json (hunk 4 above)
Comment (GunZi200) [reply boards/nucleo_l412rb_p.json:29]: Zephyr should follow soon: https://github.com/zephyrproject-rtos/zephyr/pull/33976 :) Not sure how easy it is to add for the other platforms

**Comment by valeros** [comment]
Many thanks!

**Pull request merged**